frontend/controller/actions/group.js

```diff
@@ -361,6 +366,11 @@ export default (sbp('sbp/selectors/register', {
           console.error(`[gi.actions/group/join] Error while sending key request for ${params.contractID}:`, e)
           throw e
         })
+
+        if (params.blockOriginatingContract !== false) {
+          await keyRequestPromise
+        }
+
```

Comment:
If you recall, `gi.actions/group/join` used to take some other parameter that we fought hard to get rid of in order to simplify the developer-facing API for joining a group — to avoid forcing developers to think about internal details.

Is it possible to do the same with this `blockOriginatingContract` parameter? Is there a way to get rid of it?

Reply:
That's needed to avoid a deadlock depending on which queue is being used. If it weren't there, the `identity/joinGroup` side-effect would result in a deadlock. On the other hand, usually you want to `await` to have feedback on when the operation completed.

Reply:
So there's clearly a reason for it... but the question was whether or not it's possible to do things some other way so that this parameter doesn't need to be passed in?

Reply:
Right, I guess it's always _possible_, but not sure how _feasible_ it is. Paraphrasing the reason, without this parameter:

* Either we don't block on the originating contract (identity contract), which sometimes you want to do, if it's possible as this way you get feedback
* Or we block on the originating contract, in which we can't await on this when it's called from the identity side-effect

Removing this involves one of these tradeoffs, or perhaps rewriting `join` in some way. Perhaps the option of *not* awaiting on it from the side-effect, possibly along with a `preSendCheck` hook might work, but I don't think this reduces complexity, just merely hides it (like, you still need to know that calling / not calling it is appropriate and whether or not `await` can be used and why)

Reply:
Ok, this might have to be something I return to after my break. 🔖 bookmarking for later